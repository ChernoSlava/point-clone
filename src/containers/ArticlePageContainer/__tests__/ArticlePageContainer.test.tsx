import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import { ArticlePageContainer } from "..";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    state: {
      articleTitle: "Test Title",
      image: "test-image.jpg",
      description: "Test Description",
    },
  }),
}));
describe("ArticlePageContainer tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders ArticlePage with correct props", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <ArticlePageContainer />
      </MemoryRouter>
    );

    const imgElement = screen.getByRole("img", { name: "Big ico" });
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", "test-image.jpg");

    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Test Description")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
    unmount();
  });
});
