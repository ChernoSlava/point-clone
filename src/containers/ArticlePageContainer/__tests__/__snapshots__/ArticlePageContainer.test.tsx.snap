// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePageContainer tests renders ArticlePage with correct props 1`] = `
<div>
  <div
    class="sc-kFCroH cdOwRX"
  >
    <h2
      class="sc-irLvIq kpUjiu"
    >
      Test Title
    </h2>
    <p
      class="sc-csKJxZ fyRViO"
    >
      Test Description
    </p>
    <img
      alt="Big ico"
      class="sc-eTNRI ivJwRw"
      src="test-image.jpg"
    />
    <p
      class="sc-brSamD kRpsRk"
    >
      Шикарные новости для вас :D
    </p>
    <a
      href="/"
      style="text-decoration: none; color: blue;"
    >
      Назад
    </a>
  </div>
</div>
`;
